packages/typespec-python/test/generated/encode-bytes/encode/bytes/operations/_operations.py

```diff
@@ -1631,7 +1631,7 @@ def default(self, **kwargs: Any) -> bytes:
         if _stream:
             deserialized = response.iter_bytes()
         else:
-            deserialized = _deserialize(bytes, response.json())
+            deserialized = _deserialize(bytes, response.json(), rf=_RestField(format="base64"))
```

Comment:
why are we adding rest field to the deserialization code? i'm kind of confused

Reply:
After offline discussion, there are 2 solutions:
(1) codegen call specific deserialization function instead of common function `_deseialize` like:
`deserialized = _deserialize_bytes(bytes, response.json() # when format is base64`
(2) set `format` instead of `rf` for common function `_deserialize` like:
`deserialized = _deserialize(bytes, response.json(), format="base64"`

Of course, both need reconstruct `_deserialize`. Which one do you prefer?

Reply:
Updated: 2 is better